### PR TITLE
Add EIP-2981 to ERC721ContractMetadata

### PR DESCRIPTION
**Pull request by ryanio**
Adds EIP-2981 to ERC721ContractMetadata

### CODE REVIEW

File: src/ERC721ContractMetadata.sol [hunk 1]
```diff
@@ -176,4 +188,80 @@ contract ERC721ContractMetadata is
...
+    }
+
+    /**
+     * @notice Sets the royalty basis points out of 10_000.
+     *
+     * @param newBps The value as an integer (e.g. 500 for 5%)
+     */
+    function setRoyaltyBasisPoints(uint256 newBps) external onlyOwner {
+        // Set the new royalty percent.
+        _royaltyBps = newBps;
```
Comment (jameswenzel) [inline src/ERC721ContractMetadata.sol:212]: should add a check that bps is `<= 10_000`

File: src/ERC721ContractMetadata.sol [hunk 2]
```diff
@@ -176,4 +188,80 @@ contract ERC721ContractMetadata is
...
     }
+
+    /**
+     * @notice Sets the address to receive royalties.
+     *
+     * @param newWallet The new wallet address.
+     */
+    function setRoyaltyAddress(address newWallet) external onlyOwner {
+        // Set the new royalty address.
+        _royaltyAddress = newWallet;
```
Comment (jameswenzel) [inline src/ERC721ContractMetadata.sol:199]: might consider eg not letting this be set to 0-address

File: src/ERC721ContractMetadata.sol (hunk 1 above)
Comment (ryanio) [reply src/ERC721ContractMetadata.sol:222]: added in https://github.com/ProjectOpenSea/seadrop/pull/21/commits/2c3670c9540e6158f3f722eb46d3519cc8b716d5

File: src/ERC721ContractMetadata.sol (hunk 2 above)
Comment (ryanio) [reply src/ERC721ContractMetadata.sol:204]: added in https://github.com/ProjectOpenSea/seadrop/pull/21/commits/2c3670c9540e6158f3f722eb46d3519cc8b716d5

File: src/ERC721ContractMetadata.sol [hunk 3]
```diff
@@ -39,6 +41,12 @@ contract ERC721ContractMetadata is
     ///         for random reveals.
     bytes32 _provenanceHash;
 
+    /// @notice Track the royalty percentage basis points (out of 10_000)
+    uint256 _royaltyBps;
```
Comment (jameswenzel) [inline src/ERC721ContractMetadata.sol:45]: can pack these into a single slot by using a `uint96` or smaller

File: src/ERC721ContractMetadata.sol [hunk 4]
```diff
@@ -176,4 +198,82 @@ contract ERC721ContractMetadata is
...
+     */
+    function provenanceHash() external view override returns (bytes32) {
+        return _provenanceHash;
+    }
+
+    /**
+     * @notice Returns the address that receives royalties.
+     */
+    function royaltyAddress() external view returns (address) {
+        return _royaltyAddress;
```
Comment (jameswenzel) [inline src/ERC721ContractMetadata.sol:230]: @Slokh is there a specific function signature we're expecting to get recipient/bps or is that only for multi-recipients?

Comment (Slokh) [reply src/ERC721ContractMetadata.sol:230]: 2981 is just a single function that we use, i think these are just helpers ryan added in addition

```
    function royaltyInfo(
        uint256 _tokenId,
        uint256 _salePrice
    ) external view returns (
        address receiver,
        uint256 royaltyAmount
    );
    ```

Comment (ryanio) [reply src/ERC721ContractMetadata.sol:230]: yeah `royaltyAddress` and `royaltyBasisPoints` were just two getters I added for convenience to compliment the setters

**Review by jameswenzel — APPROVED** [review]
LGTM!

**Pull request merged**